### DISCUSSION TITLE
Can the next auto-increment ID be determined at time of record creation?

### DISCUSSION THREAD

**Discussion by simonsinclair** [prisma/prisma#7766] [Q&A]
Hello, Prisma community.

I would like to generate a unique ID based on the auto-incrementing primary key of each record in my application to use in URLs – see https://github.com/prisma/prisma/discussions/7766#discussioncomment-892061 for an example. I am looking to achieve a [similar result to Reddit](https://www.reddit.com/r/redditdev/comments/14yppc/how_does_reddit_come_up_with_unique_ids_for_things/), with their Base 36 representation of integer IDs.

What would be the Prisma-way of doing this? Is it possible to determine the next auto-increment ID at time of record creation?

**Comment by felinto-dev** [comment]
I am not sure if I understand your question.

Prisma will return for you the object that was created in the database after you save OR you can only use a findOne or findMany and order by "updatedAt" column, DESC to know what is the last index.

**Reply by simonsinclair** [reply]
Hi @felinto-dev, thanks for your reply. To clarify, I'd like to find a way of doing something like this:

```prisma
model Food {
  id   Int    @id @default(autoincrement())
  slug String
}
```
```js
const banana = await prisma.food.create({
  data: {
    slug: getSlug(id) // Where `id` is the next auto-incrementing id.
  },
})
```

**Reply by felinto-dev** [reply]
I still did not get it. When you create an object in the prism, the object ID will be returned.

Why do you need the ID?

https://www.prisma.io/docs/concepts/components/prisma-client/crud#create
![Screenshot_896](https://user-images.githubusercontent.com/32253743/122650119-7d2cdb00-d107-11eb-9ebe-262ad6d92f44.png)

**Reply by simonsinclair** [reply]
I understand that, but to generate a slug, I would like to know ID of the record before it is created.

**Reply by felinto-dev** [reply]
You know if you omit the ID parameter it will be created automatically, right?

**Reply by felinto-dev** [reply]
I was checking about that....
https://itnext.io/whats-a-slug-f7e74b6c23e0

I am not sure if is a good idea you derive the "slug" from ID.
https://github.com/simov/slugify

Try "slugify" the food "name".

**Reply by felinto-dev** [reply]
One way to know the ID before it is created, since you use auto-increment, could be:

```ts
const nextId = await this.prisma.food.count() + 1;
```

However, I don't know if it's appropriate to do this to solve this specific issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
